src/deploy.rs

```diff
@@ -77,4 +142,174 @@ impl Cmd {
         )?;
         Ok(())
     }
+
+    async fn run_against_rpc_server(&self, contract: Vec<u8>) -> Result<(), Error> {
+        let base_url = self.rpc_server_url.clone() + "/api/v1/jsonrpc";
+        // TODO: We should consider migrating the server subcommand to jsonspree
+        let client = HttpClientBuilder::default().build(base_url)?;
+        let key = parse_private_key(&self.seed_strkey)?;
+
+        // Get the account sequence number
+        let public_strkey =
+            stellar_strkey::StrkeyPublicKeyEd25519(key.public.to_bytes()).to_string();
+        let account_details: GetAccountResponse = client
+            .request("getAccount", rpc_params![public_strkey])
+            .await?;
+        // TODO: create a cmdline parameter for the fee isntead of simply using the minimum fee
+        let fee: u32 = 100;
+        let tx = build_create_contract_tx(
+            contract,
+            account_details.sequence,
+            fee,
+            &self.network_passphrase,
+            key,
+        )?;
+        let response: SendTransactionResponse = client
+            .request("sendTransaction", rpc_params![tx.to_xdr()?])
+            .await?;
+        println!("{}", response.status);
+        Ok(())
+    }
+}
+
+fn build_create_contract_tx(
+    contract: Vec<u8>,
+    sequence: i64,
+    fee: u32,
+    network_passphrase: &str,
+    key: ed25519_dalek::Keypair,
+) -> Result<TransactionEnvelope, Error> {
+    let salt = rand::thread_rng().gen::<[u8; 32]>();
+
+    let separator =
+        b"create_contract_from_ed25519(contract: Vec<u8>, salt: u256, key: u256, sig: Vec<u8>)";
+    let contract_hash = Sha256::new()
+        .chain_update(separator)
+        .chain_update(salt)
+        .chain_update(contract.clone())
+        .finalize();
+
+    let contract_signature = key.sign(&contract_hash);
+
+    let preimage = HashIdPreimageEd25519ContractId {
```

Comment:
we [discovered this usage is a bug in go PR](https://github.com/stellar/go/pull/4608/commits/940db2ebdf49dc2f63f74b8bad7a1cb6c030a4fd#diff-cf87fac554f98ded560ab782062123b33a65fdb9ce05add2e60423577e8b868eR63) causing `function_trapped` it should be wrapped with `HashIdPreimage` also.

Reply:
Thanks! I fixed it